packages/hash/integration/src/tests/model/ontology/link-type.model.test.ts

```diff
@@ -31,9 +33,8 @@ const linkTypeSchema: Omit<LinkType, "$id"> = {
 };
 
 beforeAll(async () => {
-  const testUser = await createTestUser(graphApi, "link-type-test", logger);
-
-  ownedById = testUser.entityId;
+  testUser = await createTestUser(graphApi, "link-type-test", logger);
+  testUser2 = await createTestUser(graphApi, "link-type-test", logger);
```

Comment:
Is it fine to pass in the same string here? It may be, I can't remember what `createTestUser` did, but I know that string was to avoid conflicts.

Reply:
Yes, [the `createTestUser` method](https://github.com/hashintel/hash/blob/51c3a539be68ca7bf1c7ffa8f13550e6f20efee4/packages/hash/integration/src/tests/util.ts#L109-L150) uses this argument as a prefix of the shortname. It then generates a random suffix for the shortname to prevent conflicts.

You can validate that no conflicts are introduced by these changes, as CI would fail.

Reply:
Hm, it's probably fine as is then. I remember the suffix not being drawn from a particularly large value space, so it will introduce instability if tests are reran on the same DB. This wasn't a problem for CI because the tests should be ran on a DB that hasn't seen this particular test before, however we can now have flakiness where the suffix could be the same.

Reply:
>  I remember the suffix not being drawn from a particularly large value space

If I'm reading this right I think the suffix comes from a pretty large value space (`26^6` if I'm not mistaken):

https://github.com/hashintel/hash/blob/51c3a539be68ca7bf1c7ffa8f13550e6f20efee4/packages/hash/integration/src/tests/util.ts#L98-L104

> It might be better to just add link-type-test-1 and link-type-test-2 (assuming we can put numbers in shortnames)

Wouldn't this just reduce the likelihood of a clash by half? Not sure if that's worthwhile if I'm understanding right and the current odds of a clash are `1 / (26 ^ 6)`

Reply:
> Wouldn't this just reduce the likelihood of a clash by half

It'd be zero

`link-type-test-1${suffix}` vs `link-type-test-2${suffix}`, they'd never clash.

Reply:
The clash I was referring to is between subsequent runs of the test, so `link-type-test-{suffix-from-run-2}` clashing with `link-type-test-{suffix-from-run-2}`, meaning the likelihood for a clash occurring in the test file would be `n * 1 / (26 ^ 6)` where `n` would be uses of the `link-type-test` prefix. So for this test file that would be `2 / (26 ^ 6)`.

Introducing `link-type-test-2` would not reduce the likelihood of this clash to 0, it'd be fairly similar. But because the likelihood is `1 / (26 ^ 2)`, and because this only affects subsequent runs where the database isn't wiped (something we want to address in the future anyway), I don't think this is a worthwhile practice to introduce in the test files.

But in the interest of getting this merged and resolving this thread, I've applied the suggestion in https://github.com/hashintel/hash/pull/1197/commits/ab689fad808bc9e81f01f8680a20438d4e376ea8

Reply:
Gotcha, but as I said, now that you've created 2 users there is _potential_ for the two users to clash in CI even on a free DB. Not between subsequent runs of the test.

Thanks for applying

Reply:
Ah missed that, apologies. Ideally we would remove the possibility of a clash in the same run by keeping track of the generated suffixes in an array, but let's stick with this for now.

Reply:
I mean I don't know about that. We're literally passing in a shortname here, and you're creating two users, so it makes sense to pass in two shortnames (you could replace these with `Bob` and `Alice` if you want, I just picked ones related to the tests incase we need to manually debug the DB and such)

Reply:
I assumed that we're using the suffix to associate created shortnames with the tests, rather than associating names with the user model instances that are created becuase this was named `"property-type-test-"` and not `"test-user"` or `"alice"`. I'm open to changing this though.

Reply:
Yes, that's what I was saying :D 

> I just picked ones related to the tests incase we need to manually debug the DB and such

But we're creating two users here, so we should be providing two shortnames, not the same one (hence the changes)